### DISCUSSION TITLE
Lesson 7 A question about HelperConfig.s.sol

### DISCUSSION THREAD

**Discussion by Osneue** [Cyfrin/foundry-full-course-cu#952] [Q&A] [answered]
From **1:00:47** to **1:01:45** in the video of Lesson 7, @PatrickAlphaC explained why we need add the code snippet
```
if (activeNetworkConfig.priceFeed != address(0)) {
     return activeNetworkConfig;
}
```
and the whole function like this
```
function getOrCreateAnvilConfig() public returns (NetworkConfig memory) {
        if (activeNetworkConfig.priceFeed != address(0)) {
            return activeNetworkConfig;
        }

        console.log("Deploying MockV3Aggregator");

        vm.startBroadcast();
        MockV3Aggregator priceFeed = new MockV3Aggregator(
            DECIMALS,
            INITIAL_PRICE
        );
        vm.stopBroadcast();

        NetworkConfig memory config = NetworkConfig({
            priceFeed: address(priceFeed),
            others: 0
        });
        return config;
    }
```
However, in the process of unit test, it always call `setUp()` function first, and then call one of unit test function, and then repeat this process until all tests are completed.

So of course the `activeNetworkConfig.priceFeed` always equals to zero, because we always create a new HelperConfig contract in `setUp()`.

- Do we really need this conditional judgment?
- Or it works in other scenarios?

**Comment by 0xusmanf** [comment, marked as answer]
From the perspective of tests, you are right, we don't really need this check but `HelperConfig.s.sol`'s use is not just limited to the tests, it can also be used in other scripts which is not the case for this lesson but you will see it in action in the next lessons.

**Reply by Osneue** [reply]
Got it, thank you.
